xla/service/hlo_dataflow_analysis.h

```diff
@@ -305,6 +305,10 @@ class HloDataflowAnalysis {
   bool UpdateAllGatherStartValueSet(HloInstruction* all_gather_start);
   bool UpdateAllGatherDoneValueSet(HloInstruction* all_gather_done);
   bool UpdateAllReduceDoneValueSet(HloInstruction* all_reduce_done);
+    bool UpdateCollectiveBroadcastStartValueSet(
```

Comment:
Looks like this is not called from anywhere? Bugs in dataflow analysis can later lead to incorrect buffer assignment.

Reply:
This is the likely culprit to the e2e issue I've been seeing

Reply:
This was the issue! Current state of the PR has the e2e tests working.